metabase/metabase
# Append to Model from CSV

Part of https://github.com/metabase/metabase/issues/35614

### Description

Adds toolbar menu item  for adding data to a model.

![Screen Shot 2024-01-09 at 4 50 55 PM](https://github.com/metabase/metabase/assets/30528226/21b9d040-4fc0-4cf3-bb46-52dc42b284b9)


Updates Status Notification to reflect that we're uploading to a table and not a collection.

![Screen Shot 2024-01-08 at 8 33 44 AM](https://github.com/metabase/metabase/assets/30528226/74947e9f-ec59-4f4b-bae9-a0e6bfe26591)

Shows nice errors if your schema doesn't match

![Screen Shot 2024-01-05 at 4 47 39 PM](https://github.com/metabase/metabase/assets/30528226/f6ad1c4c-4c06-43ab-b15b-51a725d726af)

![Screen Shot 2024-01-05 at 4 47 52 PM](https://github.com/metabase/metabase/assets/30528226/fc094c8b-b543-467c-bb66-5dffa73cca1c)

### How to verify

- enable CSV uploads in your metabase instance
- create a model from a CSV by uploading a csv to a collection
- open your model
- open the ... context menu and click "upload data to this model"
- select a CSV with the same schema (usually, you can upload the same CSV a second time)
- see that it adds the data
- try uploading a CSV with a different schema
- see that you get an error message explaining how the schema differs.


### Checklist
